src/main/java/com/google/api/codegen/InterfaceView.java

```diff
@@ -26,15 +27,13 @@
 public class InterfaceView implements InputElementView<Interface> {
 
   /**
-   * Gets the reachable interfaces of the model.
+   * Gets the interfaces for the apis in the service config.
    */
   @Override
   public Iterable<Interface> getElementIterable(Model model) {
     ArrayList<Interface> interfaces = new ArrayList<>();
-    for (Interface iface : model.getSymbolTable().getInterfaces()) {
-      if (iface.isReachable()) {
```

Comment:
I glanced through the code, and I'm not super sure. I think it is doing two things:

1. Making sure the interface is "referenced" directly or indirectly by the APIs in the API list
2. Not including interfaces which have visibility marked as internal

Use case 1 is basically cut out because we are going to the same data source; use case 2 is not a concern because we are already working on stripped protos. I'm not sure if there are other cases I am missing, but I'm not terribly concerned because we'll find out at generation time and we can fix it then.